### PR TITLE
Convert JObject to string for xamarin ios and andriod template parameter's body

### PR DESCRIPTION
**Pull request by pragnagopa**
Notification Hub requires template body to be a string. Update register templates method to accept template body as string or JSON. Fix for #25

### CODE REVIEW

**Comment by pragnagopa** [comment]
@fabiocav , @pvbchak  take a look?

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.Android.Test/Functional/PushFunctional.Test.cs [hunk 1]
```diff
@@ -103,6 +103,29 @@ public PushFunctional()
...
+                var parameters = new Dictionary<string, string>()
+                {
+                    {"channelUri", registrationId},
+                    {"templates", JsonConvert.SerializeObject(expectedTemplates)}
+                };
+                await this.GetClient().InvokeApiAsync("verifyRegisterInstallationResult", HttpMethod.Get, parameters);
+            }
+            finally
+            {
+                push.UnregisterAsync().Wait();
```
Comment (fabiocav) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.Android.Test/Functional/PushFunctional.Test.cs]: We should be able to await on these now.

File: sdk/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/UnitTests/PushUnit.Test.cs [hunk 2]
```diff
@@ -52,7 +51,7 @@ public PushUnit()
             NSData deviceToken = null;
             var exception = await AssertEx.Throws<ArgumentNullException>(
            () => mobileClient.GetPush().RegisterAsync(deviceToken));
-            Assert.AreEqual(exception.Message, "Argument cannot be null.\nParameter name: deviceToken");
+            Assert.AreEqual(exception.Message, "Value cannot be null.\nParameter name: deviceToken");
```
Comment (fabiocav) [inline sdk/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/UnitTests/PushUnit.Test.cs]: Can we assert on something more reliable than the exception message?

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/Functional/PushFunctional.Test.cs [hunk 3]
```diff
@@ -112,6 +112,29 @@ public PushFunctional()
...
+                {
+                     {"channelUri", TrimDeviceToken(channelUri.Description)},
+                     {"templates", JsonConvert.SerializeObject(expectedTemplates)}
+                };
+                await push.RegisterAsync(channelUri, templates);
+                await this.GetClient().InvokeApiAsync("verifyRegisterInstallationResult", HttpMethod.Get, parameters);
+            }
+            finally
+            {
+                push.UnregisterAsync().Wait();
```
Comment (fabiocav) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/Functional/PushFunctional.Test.cs]: Same as previous comment... we should be able to await on these.

**Comment by fabiocav** [comment]
@pragnagopa, I just had a couple of comments on some minor things, but this looks good!

**Comment by pvbchak** [comment]
LGTM

File: sdk/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/UnitTests/PushUnit.Test.cs (hunk 2 above)
Comment (pragnagopa) [inline sdk/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/UnitTests/PushUnit.Test.cs]: Changed to verify paramName

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/Functional/PushFunctional.Test.cs (hunk 3 above)
Comment (pragnagopa) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.iOS.Test/Functional/PushFunctional.Test.cs]: Changed to verify paramName

File: e2etest/test/Microsoft.WindowsAzure.MobileServices.Android.Test/Functional/PushFunctional.Test.cs (hunk 1 above)
Comment (pragnagopa) [inline e2etest/test/Microsoft.WindowsAzure.MobileServices.Android.Test/Functional/PushFunctional.Test.cs]: Will update once we move to vs 2015

**Pull request merged**